lib/legacy.coffee

```diff
@@ -51,15 +52,21 @@ module.exports = (ClassicLinter) ->
         suffix: textEditor.getGrammar().scopeName
       }
       return new Promise((resolve, reject) ->
-        temp.open(tmpOptions, (err, info) ->
+        temp.mkdir('AtomLinter', (err, tmpDir) ->
           return reject(err) if err
 
           try
-            fs.writeSync(info.fd, textEditor.getText())
-            fs.closeSync(info.fd)
+            tmpFile = path.join(tmpDir, path.basename(filePath))
+            fs.writeFileSync(tmpFile, textEditor.getText())
```

Comment:
I remember seeing an `io.js`'s commit where it barks when we execute a sync method as it jams the event loop for that event. I think it's best practice to use async `writeFile` here.